Mage.Tests/src/test/java/org/mage/test/cards/triggers/NeeraWildMageTest.java

```diff
@@ -0,0 +1,52 @@
+package org.mage.test.cards.triggers;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ *
+ * @author rjayz
+ */
+public class NeeraWildMageTest extends CardTestPlayerBase {
+
+    @Test
+    public void TestNeeraWildMage() {
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 4);
+
+        // Whenever you cast a spell, you may put it on the bottom of its owner's library.
+        // If you do, reveal cards from the top of your library until you reveal a nonland card.
+        // You may cast that card without paying its mana cost.
+        // Then put the rest on the bottom of your library in a random order.
+        // This ability triggers only once each turn.
+        addCard(Zone.BATTLEFIELD, playerA, "Neera, Wild Mage", 1);
+
+        addCard(Zone.HAND, playerA, "Consider");
+
+        addCard(Zone.LIBRARY, playerA, "Island");
+        addCard(Zone.LIBRARY, playerA, "Island");
+        addCard(Zone.LIBRARY, playerA, "Mental Note");
+        addCard(Zone.LIBRARY, playerA, "Island");
+
+        skipInitShuffling();
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Consider");
+        // Put Ponder on bottom of library
+        setChoice(playerA, true);
+        // Cast Mental Note
+        setChoice(playerA, true);
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        // Assert Consider was put on the bottom of the library
+        assertLibraryCount(playerA, "Consider", 1);
```

Comment:
This only checks that Consider is *somewhere* in the library, not that it is at the bottom.

I am not sure if you can check the position in the library directly, you'll have to look around and see if there's API for it.

If there isn't, a way around it would be to empty the library, and have the player draw cards, and check that after each card draw they draw the card you expect them to draw.

Reply:
I have modified the test to empty the library first, and made it more robust. Please tell me if the test case is a bit too complex.

Reply:
I don't think it's too complex.

If you wanted to simplify it you could choose not to cast the card, and instead play another 2 rounds (i.e. stop on precombat main of turn 3) so that the playerA draws the Consider that should be at the top of their 3 card library.

Then you can check that the Consider is in their hand and the library is made up of 2 Islands and a Mental Note.

Would be a little simpler since you would no longer need to know what Mental Note does.